common/planners/src/plan_insert_into.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020-2021 The Datafuse Authors.
+//
+// SPDX-License-Identifier: Apache-2.0.
+
+use std::sync::Arc;
+use std::sync::Mutex;
+
+use common_datablocks::DataBlock;
+use common_datavalues::DataSchemaRef;
+
+/// please do not keep this, this code is just for test purpose
+type BlockStream =
+    std::pin::Pin<Box<dyn futures::stream::Stream<Item = DataBlock> + Sync + Send + 'static>>;
+
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+pub struct InsertIntoPlan {
+    pub db_name: String,
+    pub tbl_name: String,
+    pub schema: DataSchemaRef,
+
+    #[serde(skip, default = "InsertIntoPlan::empty_stream")]
+    pub input_stream: Arc<Mutex<Option<BlockStream>>>
```

Comment:
Do we need to reference this stream more than once?
It seems like an `InsertIntoPlan` is consumed only once.
What about make `input_stream` just a `BlockStream`?

Reply:
Nice catch!  

To put it simply,  this `InsertIntoPlan` thing should be put away, when we are implementing the INSERT statement seriously.

But as for now, 

- `Arc` is for Clone (in a light way)

Currently, a plan should be Cloneable, as specified by PlanNode

https://github.com/datafuselabs/datafuse/blob/8563bfd4cec49c26b072bc2eff6a915e7749d3d8/common/planners/src/plan_node.rs#L30-L53


- `Mutex<Option<..>>` is for "interior-mutability"

As you mentioned, the BlockStream is supposed to be consumed only once, 

https://github.com/datafuselabs/datafuse/blob/8563bfd4cec49c26b072bc2eff6a915e7749d3d8/fusequery/query/src/datasources/remote/remote_table.rs#L81-L91

----------

I left some comments to indicate that this piece of code is just for test purposes, but apparently at the wrong places ;(

https://github.com/datafuselabs/datafuse/blob/8563bfd4cec49c26b072bc2eff6a915e7749d3d8/common/planners/src/plan_insert_into.rs#L11-L23

https://github.com/datafuselabs/datafuse/blob/8563bfd4cec49c26b072bc2eff6a915e7749d3d8/fusequery/query/src/sql/plan_parser.rs#L297-L302

Reply:
I seeeee! 
I saw the comments but did get what it means`:~~~(`